spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/kafka/KafkaAutoConfigurationUtils.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2012-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.autoconfigure.kafka;
+
+import org.springframework.retry.backoff.ExponentialBackOffPolicy;
+import org.springframework.retry.backoff.ExponentialRandomBackOffPolicy;
+import org.springframework.retry.backoff.FixedBackOffPolicy;
+import org.springframework.retry.backoff.SleepingBackOffPolicy;
+import org.springframework.retry.backoff.UniformRandomBackOffPolicy;
+
+/**
+ * Utilities class for Kafka auto-configuration.
+ *
+ * @author Tomaz Fernandes
+ * @since 2.7.0
+ */
+public final class KafkaAutoConfigurationUtils {
+
+	private KafkaAutoConfigurationUtils() {
+
+	}
+
+	static SleepingBackOffPolicy<?> createBackOffFrom(Long min, Long max, Double multiplier, Boolean isRandom) {
```

Comment:
Spring Boot doesn't do utils class like this. It looks like this is an internal detail anyway so I'd rather make that a package protected class?

Reply:
+1 from me.
This one is only for specific auto-configuration, so better to have it as a `private static` method over there.
See for example what we have in the `IntegrationAutoConfiguration` for `defaultPollerMetadata` bean definition.